test/irqbalance.bats

```diff
@@ -0,0 +1,198 @@
+}
+
+function teardown() {
+	cleanup_test
+	stop_crio
+	# see setup about why we have these conditionals
+	if [ -f "$IRQBALANCE_CONF".bkp ]; then
+		mv -v "$IRQBALANCE_CONF".bkp "$IRQBALANCE_CONF"
+	elif [ -f /tmp/.test_owns_irqbalance_conf ]; then
+		rm -f "$IRQBALANCE_CONF"
+		rm -f /tmp/.test_owns_irqbalance_conf
+	else
+		echo "error: Something went really wrong with $IRQBALANCE_CONF !"
+		exit 1
+	fi
+	if [ -f "$BANNEDCPUS_CONF".bkp ]; then
+		mv -v "$BANNEDCPUS_CONF".bkp "$BANNEDCPUS_CONF"
+	elif [ -f /tmp/.test_owns_bannedcpus_conf ]; then
+		rm -f "$BANNEDCPUS_CONF"
+		rm -f /tmp/.test_owns_bannedcpus_conf
+	else
+		echo "error: Something went really wrong with $BANNEDCPUS_CONF !"
+		exit 1
+	fi
+}
+
+@test "irqbalance cpu ban list save" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	if [ ! -f "$BANNEDCPUS_CONF" ]; then
+		echo "error: ${BANNEDCPUS_CONF} file should have been created by CRI-o"
+		return 2
+	fi
+
+	local banned_cpus
+	banned_cpus=$(cat "$BANNEDCPUS_CONF")
+
+	[ "$expected_banned_cpus" == "$banned_cpus" ]
+}
+
+@test "irqbalance cpu ban list restore - default" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	echo "IRQBALANCE_BANNED_CPUS=\"0\"" > "$IRQBALANCE_CONF"
+
+	local banned_cpus_for_conf
+	banned_cpus_for_conf=$(cat /proc/irq/default_smp_affinity)
+	echo "$banned_cpus_for_conf" > "$BANNEDCPUS_CONF"
+
+	# when
+	OVERRIDE_OPTIONS="--irqbalance-config-file ${IRQBALANCE_CONF}" start_crio
+
+	# then
+	local banned_cpus
+	banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+
+	[ "$banned_cpus_for_conf" == "$banned_cpus" ]
+}
+
+# disable restore file, check it does NOT clear the irqbalance config
+@test "irqbalance cpu ban list restore - disable" {
+	# given
+	if ! grep -Eq '^[1,3,7,f]{1,}$' /proc/irq/default_smp_affinity; then
+		skip "requires default IRQ smp affinity (not banned CPUs)"
+	fi
+	[ -f "$CONFIGLET" ] && rm -f "$CONFIGLET"
+
+	local expected_banned_cpus
+	expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+	if [ -z "$expected_banned_cpus" ]; then
+		echo "error: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}"
+		return 1
+	fi
+
+	[ -f "$BANNEDCPUS_CONF" ] && rm -f "$BANNEDCPUS_CONF"
```

Comment:
we should probably keep this (with known content) to emphasize that the file should be ignored

Reply:
If the used file does not exist it is created with the value found in "IRQBALANCE_BANNED_CPUS" which is set to "0" here.
As the explicit file to be used is set to the value of `/proc/irq/default_smp_affinity`, which is checked to not be zero, if the final check of the test passes means this file has been ignored and the one given as parameter used, so I do not see how keeping this file will help